app/forms/admin/fraud_risk_csv_upload_form.rb

```diff
@@ -0,0 +1,57 @@
+module Admin
+  class FraudRiskCsvUploadForm
+    include ActiveModel::Model
+
+    attr_accessor :file
+
+    validates :file, presence: true
+
+    validate :csv_has_required_headers, if: -> { file.present? }
+
+    validate :all_records_are_valid, if: -> { file.present? && csv_has_required_headers? }
+
+    def initialize(params = {})
+      super
+    end
+
+    def save
+      return false unless valid?
+
+      ApplicationRecord.transaction do
+        RiskIndicator.where.not(id: records.map(&:id)).destroy_all
```

Comment:
wouldn't it have been easier to delete everything then just re-add the whatevers in the CSV?
not sure if this methodology has any benefits?

Reply:
It would be a bit easier, just seemed weird to create new records when the data hasn't changed!